File: patterns/rule-strukturierung/rules-js/rule-strukturierung-gruppe.js
rules.JSRule({
  name: "Group Approach",
  triggers: [triggers.GroupStateChangeTrigger("MyGroup")],
  execute: (event) => {
    const members = items.getItem("MyGroup").members;
    if (members.some((i) => i.state === "NULL")) {
      console.warn("One of the Items is NULL");
      return;
    }
    const sum = members.filter((i) => i.state === "ON").length;
    items.getItem("Buzz").sendCommand(sum >= 2 ? "ON" : "OFF");
  }
});

File: patterns/rule-strukturierung/rules-python/rule_strukturierung_gruppe.py
from openhab import rule, Registry
from openhab.triggers import GroupStateChangeTrigger


@rule(triggers=[GroupStateChangeTrigger("MyGroup")])
class GroupApproach:
    def execute(self, module, input):
        members = Registry.getItem("MyGroup").getAllMembers()
        if any(str(i.getState()) == "NULL" for i in members):
            self.logger.warn("One of the Items is NULL")
            return
        on_count = sum(1 for i in members if str(i.getState()) == "ON")
        new_state = "ON" if on_count >= 2 else "OFF"
        Registry.getItem("Buzz").sendCommand(new_state)
        self.logger.info("on_count={} -> Buzz={}".format(on_count, new_state))
